docs/csharp/pattern-matching.md

```diff
@@ -1,268 +1,70 @@
+You can also test a variable to find a match on specific values. The following code shows one example where you test a value against all possible values declared in an enumeration:
 
-In this updated version, the `is` expression both tests the variable and assigns
-it to a new variable of the proper type. Also, notice that this version includes
-the `Rectangle` type, which is a `struct`. The new `is` expression works with
-value types as well as reference types.
+:::code language="csharp" source="snippets/patterns/Simulation.cs" ID="PerformOperation":::
 
-Language rules for pattern matching expressions help you avoid misusing
-the results of a match expression. In the example above, the variables `s`,
- `c`, and `r` are only in scope and definitely assigned when the respective
-pattern match expressions have `true` results. If you try to use either
-variable in another location, your code generates compiler errors.
+The previous example demonstrates a method dispatch based on the value of an enumeration. The final `_` case is a [*discard pattern*](language-reference/operators/patterns.md#discard-pattern) that matches all values. It handles any error conditions where the value doesn't match one of the defined `enum` values. If you omit that switch arm, the compiler warns that you haven't handled all possible input values. At runtime, the `switch` expression throws an exception if the object being examined doesn't match any of the switch arms. You could use numeric constants instead of a set of enum values. You can also use this similar technique for constant string values that represent the commands:
 
-Let's examine both of those rules in detail, beginning with scope. The variable
-`c` is in scope only in the `else` branch of the first `if` statement. The variable
-`s` is in scope in the method `ComputeAreaModernIs`. That's because each
-branch of an `if` statement establishes a separate scope for variables. However, the `if` statement
-itself doesn't. That means variables declared in the `if` statement are in the
-same scope as the `if` statement (the method in this case). This behavior isn't
-specific to pattern matching, but is the defined behavior for variable scopes
-and `if` and `else` statements.
+:::code language="csharp" source="snippets/patterns/Simulation.cs" ID="PerformStringOperation":::
 
-The variables `c` and `s` are assigned when the respective `if` statements are true
-because of the definitely assigned when true mechanism.
+The preceding example shows the same algorithm, but using string values instead of an enum. You would use this scenario if your application responds to text commands instead of a regular data format. In all these patterns, the *discard pattern* ensures that you handle and report an error for every input. The compiler helps you by making sure every possible input value is handled.
```

Comment:
Sounded like every input results in an error report
```suggestion
The preceding example shows the same algorithm, but uses string values instead of an enum. You would use this scenario if your application responds to text commands instead of a regular data format. In all these examples, the *discard pattern* ensures that you handle every input. The compiler helps you by making sure every possible input value is handled.
```